judging-backend/main.cpp

```diff
@@ -215,6 +215,69 @@ int main(int argc, char *argv[]) {
 			std::cerr << "fork failed" << std::endl;
 			return IE;
 		}
+	} else if (strncmp(argv[1], "asm", 4) == 0) {
+		run_cmd = "./" + judge_id;
+		// assemble program
+		pid_t pid = fork();
+		if (pid == 0) {
+			// child process
+			struct rlimit rlim;
+			rlim.rlim_cur = 5;
+			rlim.rlim_max = 5;
+			if (setrlimit(RLIMIT_CPU, &rlim)) {
+				std::cerr << "failed to set compiler time limit" << std::endl;
+				return IE;
+			}
+
+			std::string asm_filepath = dir + "/../../sedimentation-assembler/sedimentation";
```

Comment:
what is this...

Reply:
It's a reliable way to reference sedimentation in the submodule without needing the user to install the executable into their system.

Reply:
How can we guarantee that sedimentation-assembler/sedimentation exists? We could add a Makefile to compile the judge and sedimentation, if you're up to doing that.